src/components/search/Search.js

```diff
@@ -56,30 +86,101 @@ const Search = (props) => {
     setValue(e.target.value)
   }
 
+  useEffect(() => {
+    async function getCityName() {
+      const city = await extractCityName({
+        lat: geoLocation.latitude,
+        lng: geoLocation.longitude,
+      })
+      setCityName(city)
+    }
+    getCityName()
+    // TODO: Need to debounce this so that the server doesn't get killed
+    // See: https://usehooks.com/useDebounce/
+  }, [geoLocation])
+
+  let currentLocationEntry = {
+    place_id: null,
+    description: 'Current Location',
+    structured_formatting: {
+      main_text: 'Current Location',
+      secondary_text: `${cityName ? cityName : ''}`,
+    },
+  }
+
   const handleSelect = async (description) => {
+    if (description === 'Current Location') {
+      // Use fixed viewport around the lat and long of the current location
+      const lat = geoLocation.latitude
+      const lng = geoLocation.longitude
+
+      const viewportBounds = {
+        ne: { lat: lat + BOUND, lng: lng + BOUND },
+        sw: { lat: lat - BOUND, lng: lng - BOUND },
+      }
+      setViewport(viewportBounds)
+    }
     setValue(description, false)
     const viewportBounds = await getViewportBounds(
       descriptionToPlaceId.current[description],
     )
     setViewport(viewportBounds)
   }
 
+  const inputRef = useRef(null)
+
+  useEffect(() => {
+    if (value === '') {
+      inputRef.current.blur()
+      inputRef.current.focus()
+    }
+  }, [value])
```

Comment:
Move these hooks to the beginning of function after the current "use"s. Rules of hooks